From eb4702309abb33bfd9944385b131b4cb60742d1c Mon Sep 17 00:00:00 2001
Subject: [PATCH] kafel: Prevent bison/flex race

When doing highly parellel builds, there is a
race condition due to incomplete dependencies being
declared in the kafel Makefile. The Makefile does
not guarantee that flex and bison have fully completed,
thus causing several compiler errors.  This change
correctly marks the full chain of bison/flex dependencies.
---
 src/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index bc456dc..c9c1d78 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -65,8 +65,8 @@ parser.h parser.c: parser.y
 
 # DO NOT DELETE THIS LINE -- make depend depends on it.
 
-kafel.o: parser.h context.h policy.h expression.h syscall.h codegen.h
-kafel.o: common.h lexer.h
+kafel.o: parser.h parser.c context.h policy.h expression.h syscall.h codegen.h
+kafel.o: common.h lexer.h lexer.c
 context.o: context.h policy.h expression.h syscall.h common.h
 codegen.o: codegen.h context.h policy.h expression.h syscall.h common.h
 codegen.o: range_rules.h
@@ -74,8 +74,8 @@ expression.o: expression.h common.h
 policy.o: policy.h expression.h common.h
 range_rules.o: range_rules.h policy.h expression.h common.h syscall.h
 syscall.o: syscall.h common.h
-lexer.o: parser.h context.h policy.h expression.h syscall.h
-parser.o: parser.h context.h policy.h expression.h syscall.h lexer.h
+lexer.o: parser.h parser.c context.h policy.h expression.h syscall.h
+parser.o: parser.h parser.c context.h policy.h expression.h syscall.h lexer.h lexer.c
 syscalls/amd64_syscalls.o: syscall.h
 syscalls/arm_syscalls.o: syscall.h
 syscalls/mipso32_syscalls.o: syscall.h
-- 
2.7.4

